src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -162,6 +186,8 @@ export class Response implements IResponse {
 		} else if (isCompleteInteractiveProgressTreeData(responsePart)) {
 			this._responseParts.push(responsePart);
 			this._updateRepr(quiet);
+		} else if ('documents' in responsePart) {
+			this._usedContext = responsePart;
```

Comment:
The API would imply that it's valid to send multiple UsedContext pieces

Reply:
This occurred to me as well. I think it makes sense in future to aggregate all `IUsedContext` progress reports. What do you think?